diff --git a/node_modules/@rosen-bridge/bitcoin-rpc-scanner/dist/DogeRpcNetwork.js b/node_modules/@rosen-bridge/bitcoin-rpc-scanner/dist/DogeRpcNetwork.js
index 5d2e613..74f6457 100644
--- a/node_modules/@rosen-bridge/bitcoin-rpc-scanner/dist/DogeRpcNetwork.js
+++ b/node_modules/@rosen-bridge/bitcoin-rpc-scanner/dist/DogeRpcNetwork.js
@@ -1,6 +1,8 @@
 import axios from 'axios';
 import { AbstractNetworkConnector, } from '@rosen-bridge/scanner-interfaces';
 import { randomBytes } from 'crypto';
+import rateLimit from 'axios-rate-limit';
+
 export class DogeRpcNetwork extends AbstractNetworkConnector {
     url;
     timeout;
@@ -9,12 +11,16 @@ export class DogeRpcNetwork extends AbstractNetworkConnector {
         super();
         this.url = url;
         this.timeout = timeout;
-        this.client = axios.create({
-            baseURL: this.url,
-            timeout: this.timeout,
-            headers: { 'Content-Type': 'application/json' },
-            auth: auth,
-        });
+        // Use axios-rate-limit to limit to 3 requests per second
+        this.client = rateLimit(
+            axios.create({
+                baseURL: this.url,
+                timeout: this.timeout,
+                headers: { 'Content-Type': 'application/json' },
+                auth: auth,
+            }),
+            { maxRPS: 3 }
+        );
     }
     generateRandomId = () => randomBytes(32).toString('hex');
     /**
